pkg/package-server/provider/inmem.go

```diff
@@ -248,16 +265,13 @@ func (m *InMemoryProvider) ListPackageManifests(namespace string) (*packagev1alp
 	return manifestList, nil
 }
 
-func (m *InMemoryProvider) GetPackageManifest(namespace, name string) (*packagev1alpha1.PackageManifest, error) {
-	m.mu.RLock()
-	defer m.mu.RUnlock()
+func (m *InMemoryProvider) Subscribe() (<-chan packagev1alpha1.PackageManifest, <-chan packagev1alpha1.PackageManifest, <-chan packagev1alpha1.PackageManifest, error) {
+	add := make(chan packagev1alpha1.PackageManifest)
+	modify := make(chan packagev1alpha1.PackageManifest)
+	delete := make(chan packagev1alpha1.PackageManifest)
+	m.add = append(m.add, add)
```

Comment:
I think we need to synchronize appending to these slices.

Reply:
[append is not a thread-safe operation](https://stackoverflow.com/questions/44152988/append-not-thread-safe)

Reply:
Yeah, I just discovered this when [running `go test -race` using my `FakeProvider`](https://github.com/operator-framework/operator-lifecycle-manager/pull/476/files#diff-8bcb7f56c2a87a0aecaed913c951245fR44).